src/lib/utils.py

```diff
@@ -0,0 +1,209 @@
+import os
+
+from os import path
+from os.path import join as opj
+import pandas as pd
+from typing import List
+
+
+def participants_tsv():
+
+    participants_tsv = opj(directories()["exp"], "participants.tsv")
+
+    return pd.read_csv(participants_tsv, sep="\t")
+
+
+def load_config():
+
+    config_file = opj(
+        path.dirname(path.realpath(__file__)), "analysis_pipelines_config.tsv"
+    )
+
+    return pd.read_csv(config_file, sep="\t")
+
+
+def return_team_config(team_ID: str) -> dict:
+    """
+    Args:
+        team_ID (str):
+
+    Returns:
+        dict: configuration for analysis for this team
+
+    Partly read from ``src/lib/analysis_pipelines_config.tsv``
+
+    keys:
+
+    - ``teamID``
+    - ``n_participants``
+    - ``excluded_participants``
+    - ``func_fwhm``
+    - ``directories`` see the ``directories`` function
+    - ``subject_list``: List[str] of the subject labels after filtering
+      the ``excluded_participants``
+
+    """
+
+    config = load_config()
+
+    config = config[config["teamID"] == team_ID]
+
+    if not config.empty:
+        config = config.to_dict("records")[0]
+        config["excluded_participants"] = config["excluded_participants"].split(", ")
+    else:
+        config = {
+            "teamID": None,
+            "n_participants": 108,
+            "excluded_participants": None,
+            "func_fwhm": None,
+        }
+
+    config["subject_list"] = return_subjects_list(
+        team_ID, config["excluded_participants"]
+    )
+
+    config["directories"] = directories(team_ID)
+
+    # ensure that we have the number of subjects used by the original team
+    assert len(config["subject_list"]) == config["n_participants"]
+
+    return config
+
+
+def return_subjects_list(team_ID=None, excluded_participants=None) -> List[str]:
+    """
+
+    Args:
+        team_ID (str, optional): Defaults to None.
+
+    Returns:
+        List[str]: subjects labels list
+    """
+
+    tmp = directories(team_ID)
+
+    dir_list = os.listdir(tmp["exp"])
+
+    if team_ID is None or excluded_participants is None:
+        return [dirs[-3:] for dirs in dir_list if dirs[:3] == "sub"]
+    else:
+        return [
+            dirs[-3:]
+            for dirs in dir_list
+            if dirs[:3] == "sub" and dirs[4:] not in excluded_participants
+        ]
+
+
+def directories(team_ID: str) -> dict:
+    """
+    Args:
+        team_ID (str)
+
+    Returns:
+        dict: dictionary of directories
+
+    keys:
+
+    - ``exp``: directory where the ds001734-download repository is stored
+    - ``resulr``: directory where the intermediate and final repositories will be stored
+    - ``working``: name of the directory where intermediate results will be stored
+    - ``output``: name of the directory where final results will be stored
+
+    """
```

Comment:
This should prevent contributors from having to have this in each of their notebook